File: src/pos_app/lib/core/models/branch.dart
import 'localized_text.dart';

class Branch {
  final int id;
  final LocalizedText name;
  final LocalizedText? address;
  final String? phone;

  /// Printed on receipts, when the branch has one
  final String? taxNumber;

  /// The line under the receipt; the till's own thank-you when empty
  final LocalizedText? receiptFooter;
  final bool isActive;
  final int displayOrder;
  final String dayStartTime;
  final String dayEndTime;
  final bool isOrderingEnabled;
  final bool isReservationsEnabled;

  const Branch({
    required this.id,
    required this.name,
    this.address,
    this.phone,
    this.taxNumber,
    this.receiptFooter,
    required this.isActive,
    required this.displayOrder,
    this.dayStartTime = '17:00',
    this.dayEndTime = '05:00',
    this.isOrderingEnabled = true,
    this.isReservationsEnabled = true,
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['id'] as int,
      name: LocalizedText.parse(json['name']),
      address: LocalizedText.parseNullable(json['address']),
      phone: json['phone'] as String?,
      taxNumber: json['taxNumber'] as String?,
      receiptFooter: LocalizedText.parseNullable(json['receiptFooter']),
      isActive: json['isActive'] as bool? ?? true,
      displayOrder: json['displayOrder'] as int? ?? 0,
      dayStartTime: json['dayStartTime'] as String? ?? '17:00',
      dayEndTime: json['dayEndTime'] as String? ?? '05:00',
      isOrderingEnabled: json['isOrderingEnabled'] as bool? ?? true,
      isReservationsEnabled: json['isReservationsEnabled'] as bool? ?? true,
    );
  }
}

File: src/Branch.API/Model/Branch.cs
namespace Chillax.Branch.API.Model;

public class Branch
{
    public int Id { get; set; }
    public LocalizedText Name { get; set; } = new();
    public LocalizedText? Address { get; set; }
    public string? Phone { get; set; }

    /// <summary>The tax registration number printed on receipts.</summary>
    public string? TaxNumber { get; set; }

    /// <summary>The line under the receipt, in both languages; the till's own thank-you when empty.</summary>
    public LocalizedText? ReceiptFooter { get; set; }
    public bool IsActive { get; set; } = true;
    public int DisplayOrder { get; set; }
    public TimeOnly DayStartTime { get; set; } = new(17, 0);
    public TimeOnly DayEndTime { get; set; } = new(5, 0);
    public bool IsOrderingEnabled { get; set; } = true;
    public bool IsReservationsEnabled { get; set; } = true;

    /// <summary>
    /// Ordering to a table needs an account here: a guest may still browse,
    /// but only a signed-in customer can put an order on a table. Off by
    /// default; a branch turns it on when strangers with a table's link
    /// become a problem. Ordering enforces it from its projection.
    /// </summary>
    public bool RequireSignInForTableOrders { get; set; }
}
